src/main/java/org/pf4j/update/FileVerifiers.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2012-present the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pf4j.update;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.invoke.MethodHandles;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Collection of standard file verifiers.
+ * Other implementations could e.g. be Antivirus checks, scanning for forbidden jar files,
+ * checking digital signatures etc.
+ */
+public class FileVerifiers {
```

Comment:
How I see things related to this `FileVerifier`:
- a new package `verifier`
- move `Sha512SumVerifier` class in `verifier` package
- move `BasicVerifier` class in `verifier` package
- remove `FileVerifiers` and create in his place a `CompoundFileVerifier` that implements `FileVerifier` and contain a list with verifiers. In `verify` we have the logic with `for each verifier do ...` (it's the same pattern used in PF4J - for example https://github.com/pf4j/pf4j/blob/master/pf4j/src/main/java/org/pf4j/CompoundPluginLoader.java). We can put BasicVerifier and Sha512SumVerifier as default in the list with verifiers from CompoundFileVerifier.
- in UpdateManager use a single FileVerifier (CompoundFileVerifier)
- in UpdateManager remove entirely `verifyFile` method because now we have a simple line in `downloadPlugin`
```
fileVerifier.verify(....);
```

Reply:
Yep, simplifies things and will be consistent with existing code.